<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1 Introduction to the Trio Class • BenchHub</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="1 Introduction to the Trio Class">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">BenchHub</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/v01_intro_trio.html">1 Introduction to the Trio Class</a></li>
    <li><a class="dropdown-item" href="../articles/v02_Evaluation_using_Trio.html">2 Evaluation using TrioR</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/SydneyBioX/BenchHub/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>1 Introduction to the Trio Class</h1>
            
            <h4 data-toc-skip class="date">11 March 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SydneyBioX/BenchHub/blob/main/vignettes/v01_intro_trio.Rmd" class="external-link"><code>vignettes/v01_intro_trio.Rmd</code></a></small>
      <div class="d-none name"><code>v01_intro_trio.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># devtools::load_all()</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sydneybiox.github.io/BenchHub/">BenchHub</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">DOSE</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DO.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h2>
<p>BenchHub is an R ecosystem built to make benchmarking easier.</p>
<p>It organizes data, evaluation metrics, gold standards (auxiliary
data), and even provides built-in visualization tools to help interpret
results. With BenchHub, researchers can quickly compare new methods,
gain insights, and produce trustworthy their benchmarking studies.</p>
</div>
<div class="section level2">
<h2 id="creating-trio-object">Creating Trio object<a class="anchor" aria-label="anchor" href="#creating-trio-object"></a>
</h2>
<p>Trio objects can be created using the <code>Trio$new</code>
constructor. There are 3 ways to create a Trio object:</p>
<ul>
<li>Curated Trio Datasets</li>
<li>Source and ID</li>
<li>Load an object directly</li>
</ul>
<p>If the dataset can’t be loaded using Trio’s inbuilt loader, a custom
loader can be provided.</p>
<div class="section level3">
<h3 id="method-1-curated-trio-datasets">Method 1: Curated Trio Datasets<a class="anchor" aria-label="anchor" href="#method-1-curated-trio-datasets"></a>
</h3>
<p>You can directly use the name from the Curated Trio Datasets sheet to
initialise a Trio object populated with some metrics and auxiliary data.
This method is useful when you want to quickly start with a predefined
dataset.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tempCache</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">trio</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/Trio.html">Trio</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="st">"Veteran_data"</span>, cachePath <span class="op">=</span> <span class="va">tempCache</span><span class="op">)</span></span>
<span></span>
<span><span class="va">trio</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Trio Object</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Dataset </span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset Details</span>:</span></span>
<span><span class="co">##   spc_tbl_ [137 × 9] (S3: spec_tbl_df/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   - attr(*, "spec")=</span></span>
<span><span class="co">##   .. cols(</span></span>
<span><span class="co">##   ..  ...1 = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  trt = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  celltype = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  time = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  status = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  karno = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  diagtime = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ... (truncated)</span></span>
<span><span class="co">## <span style="font-weight: bold;">Data Source</span>: <span style="color: #0000BB;">"figshare"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset ID</span>: <span style="color: #0000BB;">"26142922/47361073"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Cache Path</span>: <span style="color: #0000BB;">"/tmp/RtmpDYeaqz"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Split Indices</span>: <span style="color: #0000BB;">"None"</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Auxilliary Data </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Auxiliary Data</span>: <span style="color: #0000BB;">1</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Auxiliary Data</span>: <span style="color: #0000BB;">"survival_data"</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Metrics </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Metrics</span>: <span style="color: #0000BB;">6</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Metrics</span>: <span style="color: #0000BB;">"harrell_cindex"</span>, <span style="color: #0000BB;">"begg_cindex"</span>, <span style="color: #0000BB;">"uno_cindex"</span>, <span style="color: #0000BB;">"gh_cindex"</span>,</span></span>
<span><span class="co">## <span style="color: #0000BB;">"brier_score"</span>, and <span style="color: #0000BB;">"time_dep_auc"</span></span></span></code></pre>
<p>The above output shows that we have a Trio with a dataset, metrics,
and auxiliary data. The dataset contains 137 rows and 9 columns, and the
metrics and auxiliary data are already populated and printed.</p>
<p>This Trio is ready for use in survival prediction evaluation.</p>
</div>
<div class="section level3">
<h3 id="method-2-source-and-id">Method 2: Source and ID<a class="anchor" aria-label="anchor" href="#method-2-source-and-id"></a>
</h3>
<p>Trio objects can be created by specifying an ID from a source with a
valid trio downloader. This method is useful when you have a specific
dataset ID from a supported source like Figshare, GEO, or
ExperimentHub.</p>
<p>For example, if you have a dataset ID from Figshare, GEO, or
ExperimentHub, you can create a Trio object as follows:</p>
<ul>
<li>figshare: <code>Trio$new("figshare:figshareID[/fileID]")</code>
<ul>
<li>
<code>fileID</code> can optionally be provided to specify a specific
file in the collection.</li>
</ul>
</li>
<li>GEO: <code>Trio$new("geo:GSEID[/Supplementary_filename]")</code>
<ul>
<li>
<code>Supplementary_filename</code> can optionally be provided to
specify a specific supplementary file in the series.</li>
</ul>
</li>
<li>experiementhub:
<code>Trio$new("experiementhub:experimenthubID")</code>
</li>
</ul>
<p>The example below shows how to create a Trio object using a Figshare
dataset with a <code>fileID</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trioA</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/Trio.html">Trio</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="st">"figshare:26142922/47361079"</span>, cachePath <span class="op">=</span> <span class="va">tempCache</span><span class="op">)</span></span>
<span></span>
<span><span class="va">trioA</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Trio Object</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Dataset </span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset Details</span>:</span></span>
<span><span class="co">##   spc_tbl_ [58 × 19,820] (S3: spec_tbl_df/tbl_df/tbl/data.frame)</span></span>
<span><span class="co">##   - attr(*, "spec")=</span></span>
<span><span class="co">##   .. cols(</span></span>
<span><span class="co">##   ..  ...1 = <span style="color: #BB0000;">col_character()</span>,</span></span>
<span><span class="co">##   ..  A1BG = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  `A1BG-AS1` = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  A1CF = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  A2M = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  `A2M-AS1` = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ..  A2ML1 = <span style="color: #00BB00;">col_double()</span>,</span></span>
<span><span class="co">##   ... (truncated)</span></span>
<span><span class="co">## <span style="font-weight: bold;">Data Source</span>: <span style="color: #0000BB;">"figshare"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset ID</span>: <span style="color: #0000BB;">"26142922/47361079"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Cache Path</span>: <span style="color: #0000BB;">"/tmp/RtmpDYeaqz"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Split Indices</span>: <span style="color: #0000BB;">"None"</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Auxilliary Data </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Auxiliary Data</span>: <span style="color: #0000BB;">0</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Auxiliary Data</span>:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Metrics </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Metrics</span>: <span style="color: #0000BB;">0</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Metrics</span>:</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="method-3-load-an-object-directly">Method 3: Load an object directly<a class="anchor" aria-label="anchor" href="#method-3-load-an-object-directly"></a>
</h3>
<p>Trio can also be created by passing an object directly into the
constructor. This method is useful when you already have a dataset
loaded in your R environment and want to use it with Trio.</p>
<p>If you have your own dataset, you can easily create a trio object as
well. Below is an example using a microbiome dataset.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"lubomski_microbiome_data"</span>, package <span class="op">=</span> <span class="st">"BenchHub"</span><span class="op">)</span></span>
<span><span class="va">trioB</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/Trio.html">Trio</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>data <span class="op">=</span> <span class="va">lubomPD</span>, datasetID <span class="op">=</span> <span class="st">"lubomski_microbiome"</span><span class="op">)</span></span>
<span><span class="va">trioB</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Trio Object</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Dataset </span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset Details</span>:</span></span>
<span><span class="co">##   Factor w/ 2 levels "0","1": 2 1 2 1 2 2 1 2 2 1 ...</span></span>
<span><span class="co">## <span style="font-weight: bold;">Data Source</span>:</span></span>
<span><span class="co">## <span style="font-weight: bold;">Dataset ID</span>: <span style="color: #0000BB;">"lubomski_microbiome"</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Cache Path</span>:</span></span>
<span><span class="co">## <span style="font-weight: bold;">Split Indices</span>: <span style="color: #0000BB;">"None"</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Auxilliary Data </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Auxiliary Data</span>: <span style="color: #0000BB;">0</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Auxiliary Data</span>:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## ── Metrics </span></span>
<span><span class="co">## <span style="font-weight: bold;">Number of Metrics</span>: <span style="color: #0000BB;">0</span></span></span>
<span><span class="co">## <span style="font-weight: bold;">Names of Metrics</span>:</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="bonus-using-a-custom-loader">Bonus: Using a custom loader<a class="anchor" aria-label="anchor" href="#bonus-using-a-custom-loader"></a>
</h3>
<p>Trio supports custom loaders for data formats not directly supported
by Trio. A loader is any function that takes in a path, provided by a
downloader, and returns an object to be loaded into Trio.</p>
<p>Below, we use an anonymous function to wrap
<code><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">GEOquery::getGEO</a></code> and provide it with the path of the
downloaded file.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This code will not run as it requires a GEO dataset to be downloaded</span></span>
<span><span class="co"># GEO is unreliable</span></span>
<span><span class="va">trioGEO</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/Trio.html">Trio</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span></span>
<span>  <span class="st">"GEO:GSE46474"</span>,</span>
<span>  dataLoader <span class="op">=</span> \<span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span>filename <span class="op">=</span> <span class="va">path</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  cachePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">trioGEO</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="adding-components-to-trio">Adding components to Trio<a class="anchor" aria-label="anchor" href="#adding-components-to-trio"></a>
</h2>
<div class="section level3">
<h3 id="adding-metrics">Adding metrics<a class="anchor" aria-label="anchor" href="#adding-metrics"></a>
</h3>
<p>In benchmarking studies, a metric refers to the measurement used to
evaluate a specific task. In this example, we define a task called
survival model prediction.</p>
<p>In Trio, a metric is any pairwise function of the form
<code>f(expected, predicted)</code> which returns a single value.</p>
<p>We can also add metrics that have additional arguments by passing a
list of arguments to the args parameter.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eq</span> <span class="op">&lt;-</span> \<span class="op">(</span><span class="va">expected</span>, <span class="va">predicted</span>, <span class="va">inequality</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">inequality</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="op">!</span><span class="va">expected</span> <span class="op">==</span> <span class="va">predicted</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span>  <span class="va">expected</span> <span class="op">==</span> <span class="va">predicted</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">trio</span><span class="op">$</span><span class="fu">addMetric</span><span class="op">(</span><span class="st">"equality"</span>, <span class="va">eq</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Trio also supports passing through arguments to a metric</span></span>
<span><span class="co"># Note: parameter names added for clarity</span></span>
<span><span class="va">trio</span><span class="op">$</span><span class="fu">addMetric</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"inequality"</span>, metric <span class="op">=</span> <span class="va">eq</span>, args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>inequality <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>In the above example, we added two metrics based on the same
function: “equality” and “inequality”. The “equality” metric checks if
the expected and predicted values are equal, while the “inequality”
metric checks if they are not equal.</p>
<p>Underneeth the hood, Trio creates a wrapper function that calls the
metric function with the specified arguments.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trio</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inequality</span></span></code></pre></div>
<pre><code><span><span class="co">## function (auxData, to_eval) </span></span>
<span><span class="co">## {</span></span>
<span><span class="co">##     do.call(metric, append(list(auxData, to_eval), args))</span></span>
<span><span class="co">## }</span></span>
<span><span class="co">## &lt;environment: 0x5635d788c4b0&gt;</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="adding-and-getting-auxdata">Adding and getting <code>auxData</code><a class="anchor" aria-label="anchor" href="#adding-and-getting-auxdata"></a>
</h3>
<p><code>auxData</code> (Auxiliary Data) refers to gold standard in
evaluation tasks and falls into two categories:</p>
<ul>
<li>Internal: Extracted directly from the Trio dataset. This may include
metadata such as cell types (e.g., B cells, CD4, CD8), patient
conditions (e.g., healthy, disease), and spatial domain information. The
specific content depends on the dataset itself.</li>
<li>External: Retrieved from external databases, such as disease pathway
databases or gene marker databases.</li>
</ul>
<p>With the <code>equality</code> and <code>inequality</code> metrics
added above, we can now run an evaluation.</p>
<p>First we add some auxiliary data to the Trio object to evaluate
against.</p>
<p>Auxiliary data is any data that allows you to make an evaluation.
Auxiliary data is added linked to metrics that allow you to evaluate
it.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># It can be a fixed quantity</span></span>
<span><span class="va">trio</span><span class="op">$</span><span class="fu">addAuxData</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"Number One"</span>, auxData <span class="op">=</span> <span class="fl">1</span>, metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"equality"</span>, <span class="st">"inequality"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Or it can be a function that acts on the data</span></span>
<span><span class="va">trio</span><span class="op">$</span><span class="fu">addAuxData</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"length"</span>, auxData <span class="op">=</span> <span class="va">length</span>, metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"equality"</span>, <span class="st">"inequality"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>To view the auxiliary data one can use <code>getAuxData</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trio</span><span class="op">$</span><span class="fu">getAuxData</span><span class="op">(</span><span class="st">"length"</span><span class="op">)</span> <span class="co"># This is equivalent to `length(trio$data)`</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 9</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="evaluation-example">Evaluation Example<a class="anchor" aria-label="anchor" href="#evaluation-example"></a>
</h3>
<p>Once Trio is set up with relevant metrics and auxData, we are ready
to evaluate.</p>
<p>For simple evaluation:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="va">trio</span><span class="op">$</span><span class="fu">evaluate</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Number One"</span> <span class="op">=</span> <span class="fl">1</span>, length <span class="op">=</span> <span class="fl">182</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">datasetID</th>
<th align="left">auxData</th>
<th align="left">metric</th>
<th align="left">result</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">Number One</td>
<td align="left">equality</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">Number One</td>
<td align="left">inequality</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">length</td>
<td align="left">equality</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">length</td>
<td align="left">inequality</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<p>For multi-method evaluation:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="va">trio</span><span class="op">$</span><span class="fu">evaluate</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  great_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Number One"</span> <span class="op">=</span> <span class="fl">1</span>, length <span class="op">=</span> <span class="fl">575</span><span class="op">)</span>,</span>
<span>  poor_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Number One"</span> <span class="op">=</span> <span class="fl">2</span>, length <span class="op">=</span> <span class="fl">182</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">datasetID</th>
<th align="left">method</th>
<th align="left">auxData</th>
<th align="left">metric</th>
<th align="left">result</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">great_method</td>
<td align="left">Number One</td>
<td align="left">equality</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">great_method</td>
<td align="left">Number One</td>
<td align="left">inequality</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">great_method</td>
<td align="left">length</td>
<td align="left">equality</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">great_method</td>
<td align="left">length</td>
<td align="left">inequality</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">poor_method</td>
<td align="left">Number One</td>
<td align="left">equality</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">poor_method</td>
<td align="left">Number One</td>
<td align="left">inequality</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">26142922/47361073</td>
<td align="left">poor_method</td>
<td align="left">length</td>
<td align="left">equality</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">26142922/47361073</td>
<td align="left">poor_method</td>
<td align="left">length</td>
<td align="left">inequality</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<p>For a more detailed look at evaluation using Trio, please refer to
vignette 2.</p>
</div>
</div>
<div class="section level2">
<h2 id="other-features">Other Features<a class="anchor" aria-label="anchor" href="#other-features"></a>
</h2>
<div class="section level3">
<h3 id="caching">Caching<a class="anchor" aria-label="anchor" href="#caching"></a>
</h3>
<p>Trio uses caching to avoid lengthy downloads after the first time a
data set is accessed. The <code>cachePath</code> parameter specifies the
path to the cache directory. If not specified, the cache directory
defaults to <code>~/.cache/R/TrioR/</code>.</p>
</div>
<div class="section level3">
<h3 id="data-splitting">Data Splitting<a class="anchor" aria-label="anchor" href="#data-splitting"></a>
</h3>
<p>Trio supports data splitting for cross-validation. The
<code>split</code> method splits the data into training and test sets
for cross-validation. The <code>splitIndices</code> attribute stores the
indices for each sample.</p>
<p>Indices are generated using the <code>splitTools</code> package. The
<code>split</code> method takes in the outcome variable and the number
of folds and repeats. The <code>stratify</code> parameter can be used to
stratify the outcome variable.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trio</span><span class="op">$</span><span class="fu">split</span><span class="op">(</span>y <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">137</span>, n_fold <span class="op">=</span> <span class="fl">2</span>, n_repeat <span class="op">=</span> <span class="fl">5</span>, seed <span class="op">=</span> <span class="fl">1234</span>, stratify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">trio</span><span class="op">$</span><span class="va">splitIndices</span></span></code></pre></div>
<pre><code><span><span class="co">## $Fold1.Rep1</span></span>
<span><span class="co">##  [1]   2   3   4   5   6   7   8   9  10  11  12  13  18  21  22  24  25  26  30</span></span>
<span><span class="co">## [20]  31  34  36  37  42  45  48  49  55  58  59  60  61  64  66  68  70  71  73</span></span>
<span><span class="co">## [39]  76  77  81  82  87  89  90  91  92  93  96  97  98 102 103 104 105 108 109</span></span>
<span><span class="co">## [58] 110 111 115 122 125 126 130 132 133 134 135</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold2.Rep1</span></span>
<span><span class="co">##  [1]   1  14  15  16  17  19  20  23  27  28  29  32  33  35  38  39  40  41  43</span></span>
<span><span class="co">## [20]  44  46  47  50  51  52  53  54  56  57  62  63  65  67  69  72  74  75  78</span></span>
<span><span class="co">## [39]  79  80  83  84  85  86  88  94  95  99 100 101 106 107 112 113 114 116 117</span></span>
<span><span class="co">## [58] 118 119 120 121 123 124 127 128 129 131 136 137</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold1.Rep2</span></span>
<span><span class="co">##  [1]   3   4   5   6   7   8  13  15  21  22  23  24  28  29  30  31  32  34  35</span></span>
<span><span class="co">## [20]  42  47  48  51  52  54  57  59  60  61  63  64  66  67  73  74  77  78  79</span></span>
<span><span class="co">## [39]  82  84  85  89  90  92  93  94  96  98  99 102 103 108 109 110 112 114 116</span></span>
<span><span class="co">## [58] 122 123 124 125 126 128 131 132 133 135 136</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold2.Rep2</span></span>
<span><span class="co">##  [1]   1   2   9  10  11  12  14  16  17  18  19  20  25  26  27  33  36  37  38</span></span>
<span><span class="co">## [20]  39  40  41  43  44  45  46  49  50  53  55  56  58  62  65  68  69  70  71</span></span>
<span><span class="co">## [39]  72  75  76  80  81  83  86  87  88  91  95  97 100 101 104 105 106 107 111</span></span>
<span><span class="co">## [58] 113 115 117 118 119 120 121 127 129 130 134 137</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold1.Rep3</span></span>
<span><span class="co">##  [1]   2   5   6   8   9  11  13  14  17  19  20  21  24  26  27  30  31  32  34</span></span>
<span><span class="co">## [20]  36  37  38  41  42  44  46  47  48  52  53  54  58  59  60  62  65  66  69</span></span>
<span><span class="co">## [39]  70  71  72  75  76  77  78  80  84  85  86  88  89  90  91  92  93  94  99</span></span>
<span><span class="co">## [58] 100 103 104 109 111 112 113 121 126 127 128 129</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold2.Rep3</span></span>
<span><span class="co">##  [1]   1   3   4   7  10  12  15  16  18  22  23  25  28  29  33  35  39  40  43</span></span>
<span><span class="co">## [20]  45  49  50  51  55  56  57  61  63  64  67  68  73  74  79  81  82  83  87</span></span>
<span><span class="co">## [39]  95  96  97  98 101 102 105 106 107 108 110 114 115 116 117 118 119 120 122</span></span>
<span><span class="co">## [58] 123 124 125 130 131 132 133 134 135 136 137</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold1.Rep4</span></span>
<span><span class="co">##  [1]   1   2   5   8  10  14  16  17  19  22  23  24  25  27  28  31  33  34  37</span></span>
<span><span class="co">## [20]  39  40  42  43  44  48  53  54  56  58  61  65  67  68  70  71  72  74  77</span></span>
<span><span class="co">## [39]  78  80  81  82  83  84  85  87  89  91  93  95  97 102 104 111 113 119 120</span></span>
<span><span class="co">## [58] 122 123 125 126 127 129 131 132 133 134 135</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold2.Rep4</span></span>
<span><span class="co">##  [1]   3   4   6   7   9  11  12  13  15  18  20  21  26  29  30  32  35  36  38</span></span>
<span><span class="co">## [20]  41  45  46  47  49  50  51  52  55  57  59  60  62  63  64  66  69  73  75</span></span>
<span><span class="co">## [39]  76  79  86  88  90  92  94  96  98  99 100 101 103 105 106 107 108 109 110</span></span>
<span><span class="co">## [58] 112 114 115 116 117 118 121 124 128 130 136 137</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold1.Rep5</span></span>
<span><span class="co">##  [1]   2   3   5   6   7   8   9  10  11  13  14  15  18  19  21  23  29  31  33</span></span>
<span><span class="co">## [20]  35  37  40  42  43  45  46  47  51  52  54  56  57  60  61  62  66  67  68</span></span>
<span><span class="co">## [39]  69  72  76  79  81  86  87  93  94  97 100 101 102 103 106 107 108 111 115</span></span>
<span><span class="co">## [58] 116 117 118 119 123 126 127 128 129 132 134</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Fold2.Rep5</span></span>
<span><span class="co">##  [1]   1   4  12  16  17  20  22  24  25  26  27  28  30  32  34  36  38  39  41</span></span>
<span><span class="co">## [20]  44  48  49  50  53  55  58  59  63  64  65  70  71  73  74  75  77  78  80</span></span>
<span><span class="co">## [39]  82  83  84  85  88  89  90  91  92  95  96  98  99 104 105 109 110 112 113</span></span>
<span><span class="co">## [58] 114 120 121 122 124 125 130 131 133 135 136 137</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>In this vignette, we introduced the Trio class and demonstrated how
to create a Trio object using curated datasets, source and ID, or
loading an object directly. We also showed how to add metrics and
auxiliary data to a Trio object and evaluate the performance of
different methods using these metrics and auxiliary data. We hope this
vignette helps you get started with Trio and conduct benchmarking
studies more effectively.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.30.0         </span></span>
<span><span class="co">##  [3] AnnotationFilter_1.30.0   GenomicFeatures_1.58.0   </span></span>
<span><span class="co">##  [5] GenomicRanges_1.58.0      GenomeInfoDb_1.42.3      </span></span>
<span><span class="co">##  [7] DO.db_2.9                 AnnotationDbi_1.68.0     </span></span>
<span><span class="co">##  [9] IRanges_2.40.1            S4Vectors_0.44.0         </span></span>
<span><span class="co">## [11] Biobase_2.66.0            BiocGenerics_0.52.0      </span></span>
<span><span class="co">## [13] DOSE_4.0.0                clusterProfiler_4.14.6   </span></span>
<span><span class="co">## [15] BenchHub_0.0.1            ggplot2_3.5.1            </span></span>
<span><span class="co">## [17] BiocStyle_2.34.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.3               BiocIO_1.16.0              </span></span>
<span><span class="co">##   [3] polspline_1.1.25            bitops_1.0-9               </span></span>
<span><span class="co">##   [5] ggplotify_0.1.2             tibble_3.2.1               </span></span>
<span><span class="co">##   [7] R.oo_1.27.0                 cellranger_1.1.0           </span></span>
<span><span class="co">##   [9] datawizard_1.0.1            XML_3.99-0.18              </span></span>
<span><span class="co">##  [11] rpart_4.1.24                httr2_1.1.1                </span></span>
<span><span class="co">##  [13] lifecycle_1.0.4             vroom_1.6.5                </span></span>
<span><span class="co">##  [15] lattice_0.22-6              MASS_7.3-64                </span></span>
<span><span class="co">##  [17] insight_1.1.0               backports_1.5.0            </span></span>
<span><span class="co">##  [19] magrittr_2.0.3              Hmisc_5.2-2                </span></span>
<span><span class="co">##  [21] sass_0.4.9                  rmarkdown_2.29             </span></span>
<span><span class="co">##  [23] jquerylib_0.1.4             yaml_2.3.10                </span></span>
<span><span class="co">##  [25] ggtangle_0.0.6              cowplot_1.1.3              </span></span>
<span><span class="co">##  [27] DBI_1.2.3                   RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [29] abind_1.4-8                 multcomp_1.4-28            </span></span>
<span><span class="co">##  [31] zlibbioc_1.52.0             purrr_1.0.4                </span></span>
<span><span class="co">##  [33] R.utils_2.13.0              RCurl_1.98-1.16            </span></span>
<span><span class="co">##  [35] yulab.utils_0.2.0           nnet_7.3-20                </span></span>
<span><span class="co">##  [37] TH.data_1.1-3               rappdirs_0.3.3             </span></span>
<span><span class="co">##  [39] sandwich_3.1-1              GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">##  [41] enrichplot_1.26.6           ggrepel_0.9.6              </span></span>
<span><span class="co">##  [43] tidytree_0.4.6              MatrixModels_0.5-3         </span></span>
<span><span class="co">##  [45] performance_0.13.0          pkgdown_2.1.1              </span></span>
<span><span class="co">##  [47] DelayedArray_0.32.0         codetools_0.2-20           </span></span>
<span><span class="co">##  [49] tidyselect_1.2.1            aplot_0.2.5                </span></span>
<span><span class="co">##  [51] UCSC.utils_1.2.0            farver_2.1.2               </span></span>
<span><span class="co">##  [53] matrixStats_1.5.0           base64enc_0.1-3            </span></span>
<span><span class="co">##  [55] googledrive_2.1.1           GenomicAlignments_1.42.0   </span></span>
<span><span class="co">##  [57] jsonlite_1.9.1              Formula_1.2-5              </span></span>
<span><span class="co">##  [59] survival_3.8-3              systemfonts_1.2.1          </span></span>
<span><span class="co">##  [61] tools_4.4.3                 treeio_1.30.0              </span></span>
<span><span class="co">##  [63] ragg_1.3.3                  Rcpp_1.0.14                </span></span>
<span><span class="co">##  [65] glue_1.8.0                  SparseArray_1.6.2          </span></span>
<span><span class="co">##  [67] gridExtra_2.3               xfun_0.51                  </span></span>
<span><span class="co">##  [69] MatrixGenerics_1.18.1       qvalue_2.38.0              </span></span>
<span><span class="co">##  [71] dplyr_1.1.4                 withr_3.0.2                </span></span>
<span><span class="co">##  [73] BiocManager_1.30.25         fastmap_1.2.0              </span></span>
<span><span class="co">##  [75] SparseM_1.84-2              digest_0.6.37              </span></span>
<span><span class="co">##  [77] R6_2.6.1                    gridGraphics_0.5-1         </span></span>
<span><span class="co">##  [79] textshaping_1.0.0           colorspace_2.1-1           </span></span>
<span><span class="co">##  [81] GO.db_3.20.0                RSQLite_2.3.9              </span></span>
<span><span class="co">##  [83] R.methodsS3_1.8.2           googlesheets4_1.1.1        </span></span>
<span><span class="co">##  [85] tidyr_1.3.1                 generics_0.1.3             </span></span>
<span><span class="co">##  [87] ggsci_3.2.0                 data.table_1.17.0          </span></span>
<span><span class="co">##  [89] rtracklayer_1.66.0          S4Arrays_1.6.0             </span></span>
<span><span class="co">##  [91] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [93] parameters_0.24.2           pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [95] gtable_0.3.6                blob_1.2.4                 </span></span>
<span><span class="co">##  [97] XVector_0.46.0              htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [99] bookdown_0.42               fgsea_1.32.2               </span></span>
<span><span class="co">## [101] ProtGenerics_1.38.0         scales_1.3.0               </span></span>
<span><span class="co">## [103] png_0.1-8                   ggfun_0.1.8                </span></span>
<span><span class="co">## [105] splitTools_1.0.1            knitr_1.49                 </span></span>
<span><span class="co">## [107] rstudioapi_0.17.1           tzdb_0.4.0                 </span></span>
<span><span class="co">## [109] rjson_0.2.23                reshape2_1.4.4             </span></span>
<span><span class="co">## [111] checkmate_2.3.2             nlme_3.1-167               </span></span>
<span><span class="co">## [113] curl_6.2.1                  survAUC_1.3-0              </span></span>
<span><span class="co">## [115] ggcorrplot_0.1.4.1          cachem_1.1.0               </span></span>
<span><span class="co">## [117] zoo_1.8-13                  stringr_1.5.1              </span></span>
<span><span class="co">## [119] parallel_4.4.3              foreign_0.8-88             </span></span>
<span><span class="co">## [121] restfulr_0.0.15             desc_1.4.3                 </span></span>
<span><span class="co">## [123] pillar_1.10.1               grid_4.4.3                 </span></span>
<span><span class="co">## [125] vctrs_0.6.5                 cluster_2.1.8              </span></span>
<span><span class="co">## [127] htmlTable_2.4.3             evaluate_1.0.3             </span></span>
<span><span class="co">## [129] readr_2.1.5                 Rsamtools_2.22.0           </span></span>
<span><span class="co">## [131] mvtnorm_1.3-3               cli_3.6.4                  </span></span>
<span><span class="co">## [133] compiler_4.4.3              rlang_1.1.5                </span></span>
<span><span class="co">## [135] crayon_1.5.3                rms_7.0-0                  </span></span>
<span><span class="co">## [137] plyr_1.8.9                  fs_1.6.5                   </span></span>
<span><span class="co">## [139] stringi_1.8.4               BiocParallel_1.40.0        </span></span>
<span><span class="co">## [141] munsell_0.5.1               Biostrings_2.74.1          </span></span>
<span><span class="co">## [143] lazyeval_0.2.2              bayestestR_0.15.2          </span></span>
<span><span class="co">## [145] GOSemSim_2.32.0             quantreg_6.00              </span></span>
<span><span class="co">## [147] Matrix_1.7-2                hms_1.1.3                  </span></span>
<span><span class="co">## [149] patchwork_1.3.0             bit64_4.6.0-1              </span></span>
<span><span class="co">## [151] KEGGREST_1.46.0             SummarizedExperiment_1.36.0</span></span>
<span><span class="co">## [153] dotwhisker_0.8.3            gargle_1.5.2               </span></span>
<span><span class="co">## [155] igraph_2.1.4                broom_1.0.7                </span></span>
<span><span class="co">## [157] memoise_2.0.1               bslib_0.9.0                </span></span>
<span><span class="co">## [159] ggtree_3.14.0               fastmatch_1.1-6            </span></span>
<span><span class="co">## [161] bit_4.6.0                   ggstance_0.3.7             </span></span>
<span><span class="co">## [163] ape_5.8-1                   gson_0.1.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Cabiria Liang, Sanghyun Kim, Nick Robertson, Marni Torkel, Yue Cao, Dario Strbenac, Jean Yang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
